File: python/days/day1.py
from get_data import get_data


def get_lists(data):
    list_one = []
    list_two = []
    for element in data:
        el_one, el_two = element.split("   ")
        list_one.append(el_one)
        list_two.append(el_two)
    return list_one, list_two


def count_distance(num1, num2):
    return abs(int(num1) - int(num2))


def get_distance_list(arr1, arr2):
    list_distance = []
    for i in range(len(arr1)):
        distance = count_distance(arr1[i], arr2[i])
        list_distance.append(distance)
    return list_distance


def get_similarity_list(arr1, arr2):
    list_distance = []
    for i in range(len(arr1)):
        occurrence_number = arr2.count(arr1[i])
        list_distance.append(occurrence_number * int(arr1[i]))
    return list_distance


def part1(input_data):
    list_one, list_two = get_lists(input_data)
    distance_list = get_distance_list(sorted(list_one), sorted(list_two))
    return sum(distance_list)


def part2(input_data):
    list_one, list_two = get_lists(input_data)
    return sum(get_similarity_list(list_one, list_two))


def main():
    task_data = get_data('../data/day1.txt')
    print(part1(task_data))
    print(part2(task_data))


if __name__ == '__main__':
    main()

File: js/days/day1.js
const getData = require("../index");

const data = getData("../data/day1.txt");

const getLists = (data) => {
  const listOne = [];
  const listTwo = [];
  data.forEach((element) => {
    const [elOne, elTwo] = element.split("   ");
    listOne.push(elOne);
    listTwo.push(elTwo);
  });

  return {
    listOne,
    listTwo,
  };
};

const countDistance = (num1, num2) => Math.abs(num1 - num2);

const getDistanceList = (arr1, arr2) => {
  const listDistance = [];
  for (let i = 0; i < arr1.length; i++) {
    const distance = countDistance(arr1[i], arr2[i]);
    listDistance.push(distance);
  }
  return listDistance;
};

const getSimilarityList = (arr1, arr2) => {
  const listSimilarity = [];
  for (let i = 0; i < arr1.length; i++) {
    const occurrenceNumber = arr2.filter((item) => item === arr1[i]).length;
    listSimilarity.push(occurrenceNumber * arr1[i]);
  }
  return listSimilarity;
};

const part1 = (input) => {
  const { listOne, listTwo } = getLists(input);
  return getDistanceList(listOne.sort(), listTwo.sort()).reduce(
    (acc, sum) => sum + acc,
    0
  );
};

const part2 = (input) => {
  const { listOne, listTwo } = getLists(input);
  return getSimilarityList(listOne, listTwo).reduce((acc, sum) => acc + sum, 0);
};

console.log(part1(data));
console.log(part2(data));
